graphql/src/store/query.rs

```diff
@@ -147,6 +147,7 @@ fn build_filter_from_object(
     entity: ObjectOrInterface,
     object: &BTreeMap<String, r::Value>,
 ) -> Result<Option<EntityFilter>, QueryExecutionError> {
+    // Kamil: we do iterate over the object's properties, but it's only 1 level deep.
```

Comment:
@lutter 

Given the schema:

```graphql
type Note @entity {
  id: ID!
  text: String!
  author: User! @derived(field: "id")
}
              
type User @entity {
  id: ID!
  dateOfBirth: String
  country: String
  notes: [Note!]!
}
```

and this query:

```graphql
{
  users({
    where: {
      id_in: [1,2,3],
      notes_: {
        text_contains: "graphql"
      }
    }
  }) {
    id
    country
 }
}
```

The `build_filter_from_object` won't be able to get to `notes_ -> text_contains` because it goes 1 level deep and doesn't expect nested objects.

Reply:
What I wanted to do is to allow `r::Value` to has `Object` and flatten the vector here (line 150-185 in this file).

```rust
Ok(Some(EntityFilter::And({
        object
            .iter()
            .map(|(key, value)| { ...
```

Reply:
@lutter I need you guidance here